stan/math/prim/fun/trace_gen_quad_form.hpp

```diff
@@ -63,12 +62,12 @@ inline auto trace_gen_quad_form(EigMatD&& D, EigMatA&& A, EigMatB&& B) {
 template <typename EigMatD, typename EigMatA, typename EigMatB,
           require_all_eigen_vt<std::is_arithmetic, EigMatD, EigMatA,
                                EigMatB>* = nullptr>
-inline double trace_gen_quad_form(EigMatD&& D, EigMatA&& A, EigMatB&& B) {
+inline auto trace_gen_quad_form(const EigMatD& D, const EigMatA& A, const EigMatB& B) {
```

Comment:
Why would be auto better than double?

Reply:
Mostly overthinking. If the static matrix stuff goes through one day we'll be able to have floats or longs here but yeah I can change it back